agent/doctor/healthcheckstatus.go

```diff
@@ -0,0 +1,81 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package doctor
+
+import (
+	"errors"
+	"strings"
+)
+
+const (
```

Comment:
I think we could just make these statuses string constants instead of using int32 and iota. I think that way we don't need to have all this custom json marshalling either.

Reply:
I know we have opinions on this pattern in agent codebase.  I prefer to follow the existing pattern when sending messages to TACS or FE.
https://github.com/aws/amazon-ecs-agent/blob/master/agent/api/container/container.go#L102-L111

Reply:
but if the existing pattern is not standard or good I'd argue we should differ from it and improve our code